src/globus_cli/login_manager/manager.py

```diff
@@ -313,6 +315,30 @@ def _get_gcs_info(
             raise ValueError("Internal Error! collection_id or endpoint_id is required")
         return (resolved_ep_id, epish)
 
+    def get_specific_flow_client(
+        self,
+        flow_id: uuid.UUID,
+    ) -> globus_sdk.SpecificFlowClient:
+        scope_name = f'flow_{str(flow_id).replace("-", "_")}_user'
+        self.add_requirement(
+            str(flow_id),
+            scopes=[f"https://auth.globus.org/scopes/{flow_id}/{scope_name}"],
+        )
+        self.assert_logins(str(flow_id), assume_flow=True)
+
+        authorizer = self._get_client_authorizer(
+            str(flow_id),
+            no_tokens_msg=(
+                f"Could not get login data for flow {flow_id}. "
+                f"Try login with '--flow {flow_id}' to fix."
+            ),
```

Comment:
I'm just noticing a couple of things, which I had somehow missed when we spoke about this.
1. The `assert_logins` call only passes the flow ID, so it's pretty much guaranteed to print the correct message on failure (somehow when we spoke, I got mixed up and thought we'd be passing a larger set of values to `assert_logins`). I wanted to make sure to mention that I may have made misleading statements in the past.
2. I'm not sure how the `no_tokens_msg` value is reachable

`has_login` insists that the value be present and non-null, so I don't think it's possible for this message to print? But it's not easily statically analyzable as unreachable code. There might be some worthwhile cleanup we could apply here.